### ISSUE TITLE
Property for envelope style preference

### ISSUE THREAD

**Issue by grechaw** [marklogic/entity-services#291]
This is an implementation idea to be discussed in the 9.0-3 design changes.

Introduce a new key in a model's info section

```
"envelopeStyle" : "JSON|XML|Mixed",
"preferredIndex": "path|node"
```

The first key will determine wheter an instance converter will be generated for JSON envelopes, XML envelopes, or an environment that uses both.  The differences will be in how namespaced elements are handled for database properties and search options generation.

The second key, something requested by the field, would determine whether those same functions will generated element-range-index or path-range indexes, which is the default today.

**Comment by grechaw** [comment]
The WG rejected the idea of a vocabulary item for envelope style, but think it's rather part of the API.
So my spec updates include function changes to accommodate this requirement idea.  As part of the output from es:instance-converter-generate(), these functions will be available in 9.0-3:

```
(new) instance-to-canonical-json()
instance-to-canonical-xml
(new) instnace-to-json-envelope()
(new) instance-to-xml-envelope()
instance-to-envelope()
```

**Comment by grechaw** [comment]
I sent along a query to data integration group about this item -- I think it might still be required.
